core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java

```diff
@@ -230,18 +222,92 @@
 import static io.trino.sql.tree.WindowFrame.Type.ROWS;
 import static java.lang.String.format;
 import static java.util.Collections.emptyList;
-import static java.util.Objects.requireNonNull;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertTrue;
-import static org.testng.Assert.fail;
 
 public class TestSqlParser
 {
     private static final SqlParser SQL_PARSER = new SqlParser();
 
+    public static void assertGenericLiteral(String type)
+    {
+        assertThat(expression(type + " 'abc'")).isEqualTo(new GenericLiteral(location(1, 1), type, "abc"));
+    }
+
+    public static QuerySpecification createSelect123(NodeLocation queryLocation, NodeLocation columnLocation)
+    {
+        return simpleQuerySpecification(
+                queryLocation,
+                selectList(queryLocation, new LongLiteral(columnLocation, "123")));
+    }
+
+    private static ExistsPredicate exists(Query query)
+    {
+        return new ExistsPredicate(new SubqueryExpression(query));
+    }
+
+    private static ExistsPredicate exists(NodeLocation location, Query query)
+    {
+        return new ExistsPredicate(location, new SubqueryExpression(location, query));
+    }
+
+    private static ShowStats createShowStats(QualifiedName name, List<SelectItem> selects, Optional<Expression> where)
+    {
+        return new ShowStats(
+                new TableSubquery(simpleQuery(new Select(false, selects),
+                        new Table(name),
+                        where,
+                        Optional.empty())));
+    }
+
+    private static ShowStats createShowStats(NodeLocation location, QualifiedName name, List<SelectItem> selects, Optional<Expression> where)
+    {
+        return new ShowStats(
+                Optional.of(location),
+                new TableSubquery(
+                        location,
+                        simpleQuery(location, new Select(location, false, selects),
+                                new Table(location, name),
+                                where,
+                                Optional.empty())));
+    }
+
+    private static QualifiedName makeQualifiedName(String tableName)
+    {
+        List<Identifier> parts = Splitter.on('.').splitToList(tableName).stream()
+                .map(Identifier::new)
+                .collect(toImmutableList());
+        return QualifiedName.of(parts);
+    }
+
+    private static QualifiedName makeQualifiedName(NodeLocation location, String tableName, Boolean delimited)
```

Comment:
Same with this one. It shouldn't have moved in this commit. Also, ideally, it should be moved to QueryUtil, but in a different commit.